dev/TreeView/TreeViewList.cpp

```diff
@@ -144,7 +144,24 @@ void TreeViewList::OnDrop(winrt::DragEventArgs const& e)
             }
             else
             {
-                MoveNodeInto(m_draggedTreeViewNode.get(), insertAtNode);
+                const auto draggedTreeViewNode = m_draggedTreeViewNode.get();
+
+                // In single-selection mode we are inheriting the selection logic from TreeView's internal ListView control. When removing a
+                // selected item from the ListView, that item's selection state is set to unselected. As we are first removing the dropped
+                // item from the ListView and then adding it back to the ListView (to update its position), any dropped item which was selected
+                // at the time of the drop will be unselected.
+                // To preserve the selection state of a dropped item, we save its selection state at the time of the drop and manually re-apply
+                // the saved state after the item's position in the ListView was updated.
+                const auto draggedNodeSelectionState = winrt::get_self<TreeViewNode>(draggedTreeViewNode)->SelectionState();
+
+                MoveNodeInto(draggedTreeViewNode, insertAtNode);
+
+                // The dropped item was selected at the time of the drop. After updating its position in the ListView, it is now unselected.
+                // To preserve its selection state after a completed drop operation, we select it manually again.
+                if (draggedNodeSelectionState == TreeNodeSelectionState::Selected)
+                {
+                    ListViewModel()->UpdateSelection(draggedTreeViewNode, TreeNodeSelectionState::Selected);
```

Comment:
>UpdateSelection [](start = 37, length = 15)

This means that the tree view will raise two selection changed events when we drag and drop?  You said this is the behavior that list view has, but is it wrong? should we surpress selection changed events during this operation?

Reply:
Yes, it will raise two selection changed events currently (first one for dropped item being unselected, second one for being selected). Analog to this, if you bind to the TreeViewItem.IsSelected property with binding mode = two way, you will notice it is updated twice after a drop - first time to false and second time to true.

On first thought I think we won't be able to prevent the two TreeViewItem.IsSelected updates because the first change - TreeViewItem.IsSelected = false, is, I believe, dictated to us via the ListView when we remove that item from it.

It's possible for us to surpress the SelectionChanged event as we raise it in our TreeView implementation (for example by keeping a flag which says we are currently in a drop operation and if the flag is set, don't raise it). However, assuming we cannot prevent the TreeViewItem.IsSelected property from being updated, this would create some inconsistency here: A TreeViewItem would notice its IsSelected property being updated but the TreeView would not raise a corresponding SelectionChanged event. In that case, I think we should opt for the consistency and raise the event.